sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/SecretProperties.java

```diff
@@ -322,6 +308,21 @@ private void unpackAttributes(Map<String, Object> attributes) {
         unpackId((String) attributes.get("id"));
     }
 
+    @JsonProperty(value = "id")
+    void unpackId(String id) {
+        if (id != null && id.length() > 0) {
+            this.id = id;
+            try {
+                URL url = new URL(id);
+                String[] tokens = url.getPath().split("/");
+                this.name = (tokens.length >= 3 ? tokens[2] : null);
+                this.version = (tokens.length >= 4 ? tokens[3] : null);
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
```

Comment:
should we be logging this ?

Reply:
This line code should never get hit.
But added the log for code readability.